### PR TITLE
TAKEOVER: makefiles/color: Add color functions, new attempt. #12156

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR split out the color functions from #12156 without attempting to replace all usage but allowing to use built-ini make functions instead of shell calls.

### Testing procedure

```
diff --git a/Makefile.include b/Makefile.include
index 5ddce7f8dd..1c50e6881d 100644
--- a/Makefile.include
+++ b/Makefile.include
@@ -583,7 +583,8 @@ flashfile: $(FLASHFILE)
 ifeq (,$(FLASHFILE))
   $(error FLASHFILE is not defined for this board: $(FLASHFILE))
 endif
-
+$(info $(call c_green,Color function test))
+$(info no color)
 # By default always build ELFFILE, BINFILE and FLASHFILE
 ifeq ($(RIOT_CI_BUILD),1)
   # Don't build BINFILE on the CI to save some computation time
```

Compile any application.

### Issues/PRs references

Split out from #12156

### CODE REVIEW

File: makefiles/utils/ansi_special.mk [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ANSI_ESC := # you may or may not be able to see that character in your editor
+
+# The third parameter to the subst is a tab - be careful not to replace it with
+# spaces!
+TAB := $(subst ,,	)
+
+define NEWLINE
+
+
+endef
```
Comment (leandrolanzieri) [inline makefiles/utils/ansi_special.mk:16]: These are not being used so far, right?

Comment (fjmolinas) [reply makefiles/utils/ansi_special.mk:16]: Nope, they were used in https://github.com/RIOT-OS/RIOT/pull/12156/files when replacing all occurrences of `COLOR_ECHO` and others, but I did not want to go into the trouble of that rebase ATM.

Comment (leandrolanzieri) [reply makefiles/utils/ansi_special.mk:16]: So, we are keeping them because that would be the next step?

Comment (fjmolinas) [reply makefiles/utils/ansi_special.mk:16]: Yep, to remove the `COLOR_ECHO` usage this would be handy, but I can remove if you prefer.

**Review by leandrolanzieri — APPROVED** [review]
make_color_functions

ACK, color functions work as expected, and color is disabled when `CC_NOCOLOR=1`.

**Comment by fjmolinas** [comment]
Thanks for the review!

**Pull request merged**